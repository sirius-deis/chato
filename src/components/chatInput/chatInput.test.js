import { render, screen, fireEvent } from '@testing-library/react';
import ChatInput from './chatInput';

describe('ChatInput component', () => {
  it('should match snapshot', () => {
    const { container } = render(<ChatInput chatId={1} />);
    expect(container).toMatchSnapshot();
  });
  it('should fire an event', () => {
    render(<ChatInput chatId={1} />);
    fireEvent.change(screen.getByPlaceholderText('Write a message'), {
      target: { value: 'Some message' },
    });
  });
});
