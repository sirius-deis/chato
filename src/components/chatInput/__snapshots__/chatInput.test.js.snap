// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInput component should match snapshot 1`] = `
<div>
  <form
    class="sc-eqUAAy kBaOhr"
  >
    <label
      class="sc-imWYAI gmGsXn"
    >
      <input
        class="sc-jXbUNg eixptU"
        data-testid="file"
        type="file"
      />
      <div
        class="sc-dhKdcB jtcYHp"
        data-testid="file-icon"
        size="2"
      >
        <svg
          aria-hidden="true"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.172 7l-6.586 6.586a2 2 0 102.828 2.828l6.414-6.586a4 4 0 00-5.656-5.656l-6.415 6.585a6 6 0 108.486 8.486L20.5 13"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </label>
    <input
      class="sc-aXZVg bfdQmh"
      placeholder="Write a message"
      type="text"
      value=""
    />
    <label
      class="sc-fqkvVR fbsFCX"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
        />
        <path
          d="M4.285 9.567a.5.5 0 0 1 .683.183A3.498 3.498 0 0 0 8 11.5a3.498 3.498 0 0 0 3.032-1.75.5.5 0 1 1 .866.5A4.498 4.498 0 0 1 8 12.5a4.498 4.498 0 0 1-3.898-2.25.5.5 0 0 1 .183-.683zM7 6.5C7 7.328 6.552 8 6 8s-1-.672-1-1.5S5.448 5 6 5s1 .672 1 1.5zm4 0c0 .828-.448 1.5-1 1.5s-1-.672-1-1.5S9.448 5 10 5s1 .672 1 1.5z"
        />
      </svg>
    </label>
    <button
      class="sc-dcJsrY eOIJqp"
      style="padding: 1.2rem;"
    >
      <svg
        aria-hidden="true"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 11a7 7 0 01-7 7m0 0a7 7 0 01-7-7m7 7v4m0 0H8m4 0h4m-4-8a3 3 0 01-3-3V5a3 3 0 116 0v6a3 3 0 01-3 3z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
  </form>
</div>
`;
